docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -27,644 +27,348 @@ under the License.
 
-{{< tabs "42345893-70c3-4678-a348-4c419b337eb1" >}}
-{{< tab "Java" >}}
-```java
-Properties consumerConfig = new Properties();
-consumerConfig.put(AWSConfigConstants.AWS_REGION, "us-east-1");
-consumerConfig.put(ConsumerConfigConstants.STREAM_INITIAL_POSITION, "LATEST");
+For convenience, Flink provides the following schemas out of the box:
+1. `SimpleStringSchema` and `JsonSerializationSchema`.
 
-consumerConfig.put(ConsumerConfigConstants.RECORD_PUBLISHER_TYPE, 
-    ConsumerConfigConstants.RecordPublisherType.EFO.name());
-consumerConfig.put(ConsumerConfigConstants.EFO_CONSUMER_NAME, "my-flink-efo-consumer");
+2. `TypeInformationSerializationSchema` which creates a schema based on a Flink's `TypeInformation`.
+   This is useful if the data is both written and read by Flink.
+   This schema is a performant Flink-specific alternative to other generic serialization approaches.
 
-StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+3. `GlueSchemaRegistryJsonDeserializationSchema` offers the ability to lookup the writer's schema (schema which was used to write the record)
+   in [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html). Using this, deserialization schema record will be
+   read with the schema retrieved from AWS Glue Schema Registry and transformed to either `com.amazonaws.services.schemaregistry.serializers.json.JsonDataWithSchema`
+   that represents generic record with a manually provided schema or a JAVA POJO generated by [mbknor-jackson-jsonSchema](https://github.com/mbknor/mbknor-jackson-jsonSchema).
 
-DataStream<String> kinesis = env.addSource(new FlinkKinesisConsumer<>(
-    "kinesis_stream_name", new SimpleStringSchema(), consumerConfig));
-```
-{{< /tab >}}
-{{< tab "Scala" >}}
-```scala
-val consumerConfig = new Properties()
-consumerConfig.put(AWSConfigConstants.AWS_REGION, "us-east-1")
-consumerConfig.put(ConsumerConfigConstants.STREAM_INITIAL_POSITION, "LATEST")
+   <br>To use this deserialization schema one has to add the following additional dependency:
 
-consumerConfig.put(ConsumerConfigConstants.RECORD_PUBLISHER_TYPE, 
-    ConsumerConfigConstants.RecordPublisherType.EFO.name());
-consumerConfig.put(ConsumerConfigConstants.EFO_CONSUMER_NAME, "my-flink-efo-consumer");
+{{< tabs "8c6721c7-4a48-496e-b0fe-6522cf6a5e13" >}}
+{{< tab "GlueSchemaRegistryJsonDeserializationSchema" >}}
+{{< connector_artifact flink-json-glue-schema-registry kinesis >}}
+{{< /tab >}}
+{{< /tabs >}}
 
-val env = StreamExecutionEnvironment.getExecutionEnvironment()
+4. `AvroDeserializationSchema` which reads data serialized with Avro format using a statically provided schema. It can
+   infer the schema from Avro generated classes (`AvroDeserializationSchema.forSpecific(...)`) or it can work with `GenericRecords`
+   with a manually provided schema (with `AvroDeserializationSchema.forGeneric(...)`). This deserialization schema expects that
```

Comment:
"When deserializing Avro`GenericRecord` the deserialization schema expects the records DO NOT contain an embedded schema." (or equivalent)

(AFAIK specific objects always contain the schema. Don't they?)

Reply:
I'm not sure - I just maintained this from the previous docs. Added the rephrasing